src/parts.ts

```diff
@@ -834,7 +834,8 @@ export abstract class Part<StateType> {
             plugin.update(elem!)
         })
         this.eachChild(child => {
-            child._update()
+            // if the child part is dirty, don't bother to call update on it because presumably it will be rerendered shortly
+            if (child._renderState != "dirty") child._update()
```

Comment:
I also considered this:

```suggestion
            if (child._renderState != "dirty") {
              // Child is dirty, make sure it gets fully re-rendered
              child._markClean()
            } else {
              // Child is either clean or stale, either way, update it because it had a stale ancestor
              child._update()
            }
```

But I wasn't sure if that would be too significant a change to the semantics of the `_update` function

Reply:
It seems like you have the conditional backwards in the extended version. Regardless, I'm not sure it's actually necessary. I would stick with the one-liner unless you have a specific reason to do the latter.